src/test/java/guitests/FilterTests.java

```diff
@@ -64,6 +64,91 @@ public void milestoneAliasFilterTest() {
         assertEquals(0, issuePanel.getIssueCount());
     }
 
+    @Test
+    public void countFilterTest(){
+        ListPanel issuePanel = find("#dummy/dummy_col0");
+
+        // Checking 2 issues show for count:2
+        click("#dummy/dummy_col0_filterTextField");
+        selectAll();
+        type("count:2");
+        push(KeyCode.ENTER);
+
+        assertEquals(2, issuePanel.getIssueCount());
+
+        // Checking 7 issues shown for count:7
+        click("#dummy/dummy_col0_filterTextField");
+        selectAll();
+        type("count:7");
+        push(KeyCode.ENTER);
+
+        assertEquals(7, issuePanel.getIssueCount());
+
+        // if the count is greater than the number of issues, all the issues are shown in the list view
+        click("#dummy/dummy_col0_filterTextField");
+        selectAll();
+        type("count:15");
+        push(KeyCode.ENTER);
+
+        assertEquals(10, issuePanel.getIssueCount());
+
+        // Checking 0 issues show for count:0
+        click("#dummy/dummy_col0_filterTextField");
+        selectAll();
+        type("count:0");
+        push(KeyCode.ENTER);
+
+        assertEquals(0, issuePanel.getIssueCount());
+
+        //first count qualifier chosen
+        click("#dummy/dummy_col0_filterTextField");
+        selectAll();
+        type("count:6 count:9");
+        push(KeyCode.ENTER);
+
+        assertEquals(6, issuePanel.getIssueCount());
+
+        // Not-a-number returns the whole list
+
+        click("#dummy/dummy_col0_filterTextField");
+        selectAll();
+        type("count:abcd");
+        push(KeyCode.ENTER);
+
+        assertEquals(10, issuePanel.getIssueCount());
+
+        // Not-a-number returns the whole list
+
+        click("#dummy/dummy_col0_filterTextField");
+        selectAll();
+        type("count:abcd");
+        push(KeyCode.ENTER);
+
+        assertEquals(10, issuePanel.getIssueCount());
+
+        // Return the value from the first valid count qualifier
+
+        click("#dummy/dummy_col0_filterTextField");
+        selectAll();
+        type("count:abc count:8 count:2");
```

Comment:
Okay, this may be a better way to interpret it. Right now we're just doing the simplest thing and taking the first (leftmost, if we think of the expression as a tree) valid qualifier. Same for `sort` if there are multiple qualifiers.